Advanced-Recruit-Training/src/BatteryStateDisplayService.cpp

```diff
@@ -25,9 +27,20 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
 {
     batteryStateOfChargeService_.addData(batteryData);
 
+    QString TIME_PARAMETER;
```

Comment:
Only constants should be in CAP_CASE, this is a normal variable and should be in camelCase. Also maybe try to think of a better name, it's not really a parameter and it doesn't really indicate time either.

Also for safety, you should initialize the variable. It helps the longevity of the code.
```c++
QString TIME_PARAMETER = " Time remaining until Battery is Depleted: ";
     if (batteryStateOfChargeService_.isCharging())
     {
         TIME_PARAMETER = " Time remaining until Battery is Charged: ";
     }
```